pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -0,0 +1,181 @@
+package k8s
+
+import (
+	"context"
+	"runtime/pprof"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/promutils/labeled"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/lyft/flytestdlib/contextutils"
+	"github.com/lyft/flytestdlib/promutils"
+)
+
+const resourceLevelMonitorCycleDuration = 10 * time.Second
+const KindKey contextutils.Key = "kind"
+
+// This object is responsible for emitting metrics that show the current number of a given K8s resource kind, cut by namespace.
+// It needs to be kicked off. The periodicity is not currently configurable because it seems unnecessary. It will also
+// a timer measuring how long it takes to run each measurement cycle.
+type ResourceLevelMonitor struct {
+	Scope promutils.Scope
+
+	// Meta timer - this times each collection cycle to measure how long it takes to collect the levels GaugeVec below
+	CollectorTimer *labeled.StopWatch
```

Comment:
is this a prometheus collector?

Reply:
if not have you considered using - https://godoc.org/github.com/prometheus/client_golang/prometheus#Collector

Reply:
stopwatches are defined in flytestdlib, but they're basically Summary objects https://github.com/lyft/flytestdlib/blob/c963e9ff8fc9bb46f169c43b923af07d6c5bf71e/promutils/scope.go#L72